classes/utils.php

```diff
@@ -410,4 +412,67 @@ public static function moodle_version_is(string $operator, string $version): boo
 
         return false;
     }
+
+    /**
+     * We hide 'All participants' option in group mode. It doesn't make sense to display question of all groups together,
+     * and it makes confusing in reports. If the group = 0, NULL or an invalid group,
+     * we force to chose first available group by default.
+     *
+     * @param stdClass $cm Course module class.
+     */
+    public static function set_default_group($cm) {
+        global $USER;
+
+        $allowedgroups = groups_get_activity_allowed_groups($cm, $USER->id);
+        if ($allowedgroups && !groups_get_activity_group($cm, true, $allowedgroups)) {
+            $_GET['group'] = reset($allowedgroups)->id;
```

Comment:
Directly manupulating $_GET is generally not allowed in Moodle code, and if you must to it, it certainly needs a comment to explain what it is for, and why it is unavodiable. Please add.